### DISCUSSION TITLE
How to build Cefsharp locally ?

### DISCUSSION THREAD

**Discussion by ehsan1618** [cefsharp/CefSharp#4567] [General]
I can see some C++ projects in the solution. Is there any specific way to build C++ project first

**Comment by amaitland** [comment]
You need to install the required `Visual Studio` development tools. For the current supported version you need either `VS2019` or `VS2022`.

You should be promoted in `Visual Studio` to install the required tools in the solution explorer.

The [.vsconfig](https://github.com/cefsharp/CefSharp/blob/master/.vsconfig) file is what `Visual Studio` uses to determine the components are required. You can install them manually from the command line if required.

The `build.ps1` script will build and generate the Nuget packages.

If you are having problems building then it's likely `Visual Studio` is missing required component(s).

> Is there any specific way to build C++ project first

No. They are build as part of the solution in order. They are `Visual C++` projects.

**Reply by ehsan1618** [reply]
Visual studio 2022 does not prompt to install the required tools. I can see visual c++ tools are already installed

**Reply by amaitland** [reply]
Are you building the latest stable version? Check the `.vsconfig` file and manually confirm the components are installed.

**Reply by amaitland** [reply]
`VS2019` and `VS2022` should both prompt you to install missing components (when building recent versions).

![image](https://github.com/cefsharp/CefSharp/assets/307872/93bb05ef-a341-4533-94cf-55afc7172888)
